cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -387,7 +387,8 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
         private Set<String> kafkaExternalAdvertisedHostnames = new LinkedHashSet<>();
         private Set<String> kafkaExternalAdvertisedPorts = new LinkedHashSet<>();
         private Map<Integer, Set<String>> kafkaExternalDnsNames = new HashMap<>();
-        private boolean kafkaAncillaryCmChange;
+        private Long ancillaryZkCmGeneration = 0L;
+        private Long ancillaryKafkaCmGeneration = 0L;
```

Comment:
I think `kafkaAncillaryCmGeneration` is a more natural ordering of these terms. `zkAncillaryCmGeneration` too.